textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -256,24 +273,50 @@ private ITuple joinTuples(ITuple outerTuple, ITuple innerTuple, JoinPredicate jo
 			return null;
 		}
 
-		// TODO Discuss and implement as to what Schema has to be.
-		// It shouldn't just be inner tuple or outer tuple schema.
-		Schema schema = innerTuple.getSchema();
-		List<IField> fieldList = innerTuple.getFields();
-		IField[] nextTupleField = new IField[fieldList.size()];
+		List<IField> newFieldList = new ArrayList<>();
+
+		ArrayList<Attribute> innerAttrList = new ArrayList<>();
+		innerAttrList.addAll(innerTuple.getSchema().getAttributes());
```

Comment:
"Why not just
List<Attribute> innerAttrList = innerTuple.getSchema().getAttributes();"
I will change it. Before the PR I was trying few different things and as a result had to use ArrayList.